android/src/main/java/com/amplitude/android/Amplitude.kt

```diff
@@ -53,12 +54,28 @@ open class Amplitude(
             lastEventTime = storage.read(Storage.Constants.LAST_EVENT_TIME) ?. let {
                 it.toLong()
             } ?: -1
-            add(AndroidContextPlugin())
+            androidContextPlugin = AndroidContextPlugin()
+            add(androidContextPlugin)
             add(AndroidLifecyclePlugin())
         }
         add(AmplitudeDestination())
     }
 
+    /**
+     * Reset identity:
+     *  - reset userId to "null"
+     *  - reset deviceId via AndroidContextPlugin
+     * @return the Amplitude instance
+     */
+    override fun reset(): Amplitude {
+        this.setUserId(null)
+        amplitudeScope.launch(amplitudeDispatcher) {
+            idContainer.identityManager.editIdentity().setDeviceId(null).commit()
```

Comment:
Do we need this?

Reply:
Yes, since `initializeDeviceId` will skip if `deviceId` already exists